File: string/find-maximum-number-of-string-pairs/find-maximum-number-of-string-pairs.go
package findmaximumnumberofstringpairs

func maximumNumberOfStringPairs(words []string) int {
	lookup := map[string]struct{}{}
	pair := 0
	for _, word := range words {
		if _, ok := lookup[string([]byte{word[1], word[0]})]; ok {
			pair++
		}
		lookup[word] = struct{}{}
	}
	return pair
}

File: string/find-maximum-number-of-string-pairs/find-maximum-number-of-string-pairs.ts
function maximumNumberOfStringPairs(words: string[]): number {
    const lookup = new Set<string>();
    let pairs = 0;

    for (let word of words) {
        if (lookup.has(word[1] + word[0])) {
            pairs++;
        }
        lookup.add(word)
    }

    return pairs;
};
